Panic in registry code during test

Test log with failure: https://travis-ci.org/GoogleCloudPlatform/kubernetes/jobs/47373576

Stack trace of panicing thead:
```
goroutine 135 [running]:
runtime.panic(0xb1c5a0, 0x115b893)
	/home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/panic.c:279 +0xf5
github.com/GoogleCloudPlatform/kubernetes/pkg/registry/etcd.func·007(0x7f22c05d2c08, 0xc208952900, 0x2e8)
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/registry/etcd/etcd.go:351 +0x167
github.com/GoogleCloudPlatform/kubernetes/pkg/tools.(*etcdWatcher).sendAdd(0xc20852e000, 0xc20844c000)
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/tools/etcd_tools_watch.go:285 +0x345
github.com/GoogleCloudPlatform/kubernetes/pkg/tools.(*etcdWatcher).sendResult(0xc20852e000, 0xc20844c000)
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/tools/etcd_tools_watch.go:383 +0x122
github.com/GoogleCloudPlatform/kubernetes/pkg/tools.(*etcdWatcher).translate(0xc20852e000)
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/tools/etcd_tools_watch.go:235 +0x2d2
created by github.com/GoogleCloudPlatform/kubernetes/pkg/tools.newEtcdWatcher
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/tools/etcd_tools_watch.go:155 +0x1a5

```

```
F0117 20:05:09.222292   16054 integration.go:318] Failed listing service with supplied self link '/api/v1beta1/services/selflinktest?namespace=default': 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
I0117 20:05:09.229394   16054 logs.go:40] etcd DEBUG: network error:Get http://localhost:4001/v2/keys/registry/pods?consistent=true&recursive=true&sorted=false: dial tcp: i/o timeout
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x48 pc=0x6b8377]
```

```
I0117 20:05:09.221384   16054 errors.go:52] An unchecked error was received: 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
F0117 20:05:09.222292   16054 integration.go:318] Failed listing service with supplied self link '/api/v1beta1/services/selflinktest?namespace=default': 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
I0117 20:05:09.229394   16054 logs.go:40] etcd DEBUG: network error:Get http://localhost:4001/v2/keys/registry/pods?consistent=true&recursive=true&sorted=false: dial tcp: i/o timeout
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x48 pc=0x6b8377]
```

--- a/pkg/registry/etcd/etcd.go
+++ b/pkg/registry/etcd/etcd.go
@@ -347,7 +347,17 @@ func (r *Registry) WatchControllers(ctx api.Context, label, field labels.Selecto
 		}
 		match := label.Matches(labels.Set(controller.Labels))
 		if match {
-			pods, _ := r.ListPods(ctx, labels.Set(controller.Spec.Selector).AsSelector())
+			pods, err := r.ListPods(ctx, labels.Set(controller.Spec.Selector).AsSelector())
+			if err != nil {
+				glog.Warningf("Error listing pods: %v", err)
+				// No object that's useable so drop it on the floor
+				return false
+			}
+			if pods == nil {
+				glog.Warningf("Pods list is nil.  This should never happen...")
+				// No object that's useable so drop it on the floor
+				return false
+			}
 			controller.Status.Replicas = len(pods.Items)
 		}
 		return match

